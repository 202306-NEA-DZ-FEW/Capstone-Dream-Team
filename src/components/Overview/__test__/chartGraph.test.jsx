import renderer from "react-test-renderer";

import ChartGraph from "../chartGraph";

// Mock the next-i18next useTranslation hook
jest.mock("next-i18next", () => ({
    useTranslation: () => ({ t: (key) => key }),
}));

// Mock the Firebase auth and Firestore functions
jest.mock("@/util/firebase", () => ({
    auth: {
        onAuthStateChanged: jest.fn(),
    },
    db: {
        collection: jest.fn(),
        onSnapshot: jest.fn(),
    },
}));

it("renders correctly", () => {
    const tree = renderer.create(<ChartGraph />).toJSON();
    expect(tree).toMatchSnapshot();
});
