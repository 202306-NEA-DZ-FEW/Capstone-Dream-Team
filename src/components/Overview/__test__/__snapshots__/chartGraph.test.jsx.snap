// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="mx-6 col-span-12 rounded-md border border-stroke bg-white px-5 pt-7 pb-7 shadow sm:px-7.5 xl:col-span-8"
>
  <div
    className="flex w-full flex-col flex-wrap items-start justify-between gap-3 sm:flex-nowrap"
  >
    <p
      className="font-bold text-primary"
    >
      overview.chartGraph.donationPerDay
    </p>
    <div
      className="w-full"
    >
      <div
        className="-ml-5"
      />
    </div>
  </div>
</div>
`;
